### PR TITLE
PX4 v1.14 release note

### PR DESCRIPTION
**Pull request by junwoo091400**
## About
This is the release note for the upcoming v1.14 PX4 release.

It was ported over from the excel sheet file from https://github.com/PX4/PX4-Autopilot/issues/21384

### CODE REVIEW

**Comment by junwoo091400** [comment]
Note, when we branch out for 1.14 in docs, probably we will have to switch the links to the docs in the release note into 1.14 branch, not the main branch one

**Comment by hamishwillee** [comment]
> Note, when we branch out for 1.14 in docs, probably we will have to switch the links to the docs in the release note into 1.14 branch, not the main branch one

No you won't. Just use relative links. I'm doing that now.

**Comment by junwoo091400** [comment]
> No you won't. Just use relative links

Oh true 🤦‍♂️ how did I forget this!!

File: en/releases/1.14.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+Since from Ubuntu 22.04 the Gazebo classic is not officially supported, and only the [new Gazebo](https://docs.px4.io/main/en/sim_gazebo_gz/) can be installed.
+
+Simulation support for the new Gazebo is getting updated, but still lacks diverse set of vehicles available in Gazebo classic. Please feel free to contribute more simulation targets!
+
+### ROS2 & XRCE-DDS bridge
+
+Native [ROS2 support with micro XRCE-DDS bridge](https://docs.px4.io/main/en/ros/ros2_comm.html) was added.
+
+This means that now without needing MAVROS, you can establish a direct connection to the ROS2 Data Distribution Service (DDS) through UDP, TCP, serial or other communication channels.
```
Comment (hamishwillee) [inline en/releases/1.14.md:31]: Not the point. I'll rewrite this.

File: en/releases/1.14.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+:::note
+XRCE-DDS was an experimental opt-in feature in v1.13.
+:::
+
+The new architecture provides a more seemless integration with ROS 2, using the same message definition for both uORB and ROS2 topics, which are maintained in a single shared repository.
+It also provides a building block for microROS, which will allow ROS 2 nodes to be defined directly within PX4.
+
+ROS 2 users from PX4 v1.13 will need to use the new XRCE_DDS agent, and build their applications using the new message definitions.
+However they should generally not have to update their actual ROS 2 code.
```
Comment (hamishwillee) [inline en/releases/1.14.md:50]: @beniaminopozzan Can you please check this section? I have rewritten it a bit from the original. What I am trying to capture is that when migrating you don't have to rewite your own code,.

I also want to capture the reasons we made the change. Essentially as precursor to microROS, but also to vastly simplify the dependencies - down to just the one shared message repo.

**Comment by hamishwillee** [comment]
Can we get something on RemoteID support. It is absolutely essential. Not documented - no one seems to know about it. I'm slowly trying to extract info but won't be this month.

**Comment by hamishwillee** [comment]
Oh, and thanks @junwoo091400 - good to get this in place. I'm happy to merge this once Ben has had a look, and you have had a comment about remote ID.

Any further thought on when PX4 v1.14 might actually release?

File: en/releases/1.14.md (hunk 2 above)
Comment (beniaminopozzan) [reply en/releases/1.14.md:50]: Thanks @hamishwillee , you actually have to make small changes on your code to migrate to micro XRCE-DDS:

1. the topic names pattern changed. For example, before it was
   ```sh
   fmu/sensor_combined/out
   ```
   now it is
   ```sh
   fmu/out/sensor_combined
   ```
2. You have to adjust the QoS

Therefore the migration is not so effortless unfortunately.
And no, there is not a "migration guide" :(

Apart from that, it looks good to me!

Comment (hamishwillee) [reply en/releases/1.14.md]: Thanks @beniaminopozzan . I integrated your specific examples. I think that is enough to show it won't be magic, but the changes should be constrained. A migration guide might be nice, but I think most things you need can be worked out from our existing docs

Comment (hamishwillee) [reply en/releases/1.14.md]: @beniaminopozzan Actually, what if I just put those same comments in the ROS 2 Guide under a heading "Migrating from PX4 v1.13/MicroRTPS?" WHat else would we say?

FYI I'm going to merge this now. @junwoo091400 Still need remoteID info.

Comment (beniaminopozzan) [reply en/releases/1.14.md]: @hamishwillee , I have to check and try myself upgrading an old ROS node written for MicroRTPS to the new version.
Give me a couple of days, I'll do that friday

Comment (hamishwillee) [reply en/releases/1.14.md]: Thanks @beniaminopozzan - really appreciate how well you just sort stuff out. I am starting to wonder about your work-life balance!

**Comment by junwoo091400** [comment]
> Can we get something on RemoteID support. It is absolutely essential. Not documented

I have added this to TODO to keep track & asked for a volunteer to write up the documentation in discord, thank you for pointing this out 🙏

**Comment by hamishwillee** [comment]
> > Can we get something on RemoteID support. It is absolutely essential. Not documented
> 
> I have added this to TODO to keep track & asked for a volunteer to write up the documentation in discord, thank you for pointing this out 🙏

Thanks. That won't work. I'll have to do the docs myself but you should still get high level release note in.

**Pull request merged**